terraform-aws-github-runner/modules/runners-instances/templates/install-config-runner.sh

```
@@ -265,10 +265,13 @@ fi
 
 ./config.sh --unattended --name $INSTANCE_ID --work "_work" $CONFIG
 
-# Set tag as runner id for scale down later
+# Set tag `GithubRunnerID` as runner id for scale down later
 GH_RUNNER_ID=$(jq '.agentId' .runner)
 retry aws ec2 create-tags --region $REGION --resource $INSTANCE_ID --tags "Key=GithubRunnerID,Value=$GH_RUNNER_ID"
 
+# Remove tag `Stage`` from instance to indicate that the instance is finished the previous step with fresh start
+retry aws ec2 delete-tags --region "$REGION" --resources "$INSTANCE_ID" --tags "Key=Stage"
```

@jeanschmidt  @ZainRizvi 

this makes assumption that. install-config-runner.sh will be triggered with user-data after the ec2 instance is completed replaceVolumn:
Specifically here:
https://github.com/pytorch/test-infra/blob/a2e82b980b66a736f870910316eb814652f25b42/terraform-aws-github-runner/modules/runners-instances/templates/user-data.sh#L102C3-L102C24

---

I think it should since the ephemeral runner stuff works right now without a ton of modification

---

Why do you want to retry this step? And what will happen if `Stage` tag is missing when you invoke this command?

---

if stage tag is missing, no error occurr, just silently pass